app/jobs/ingest_archival_media_bag_job.rb

```diff
@@ -66,22 +66,41 @@ def ingest
         component_groups.each do |cid, sides|
           media_resource = find_or_create_media_resource(cid)
           media_resource_change_set = MediaResourceChangeSet.new(media_resource, source_metadata_identifier: media_resource.source_metadata_identifier.first)
+          add_av(media_resource_change_set, sides)
+          add_pbcore(media_resource_change_set, sides)
+          media_resource_change_set.member_of_collection_ids += Array.wrap(collection.id)
+          changeset_persister.save(change_set: media_resource_change_set)
+        end
+      end
+
+      private
+
+        def add_av(media_resource_change_set, sides)
           sides.each do |side|
             file_set = create_av_file_set(side)
             media_resource_change_set.member_ids += Array.wrap(file_set.id)
             media_resource_change_set.sync
           end
-          media_resource_change_set.member_of_collection_ids += Array.wrap(collection.id)
-          changeset_persister.save(change_set: media_resource_change_set)
         end
-      end
 
-      private
+        def add_pbcore(media_resource_change_set, sides)
+          sides.map { |side| side.split("_").first }.uniq.each do |barcode|
+            file_set = create_pbcore_file_set(barcode)
+            media_resource_change_set.member_ids += Array.wrap(file_set.id)
```

Comment:
Why not just do `+= [file_set.id]`?

Reply:
It strikes me this is all very similar to what would happen if you did something like

```
file = IngestableFile.new(file_path: pbcore.path, mime_type: "application/xml; schema=pbcore", original_filename: pbcore.path.basename, container_attributes: {transfer_notes: my_transfer_notes})
media_resource.files = [file]
```
and then save that, relying on FileAppender.

Reply:
Hm, you're right this is how it's done for FileAppender. I modeled the a/v filesets on FileAppender initially (couldn't use it because the multiple files needed to be on one fileset with special pcdm:use and everything). And then based this on that. I'm not sure whether it's better to do it this way or use the FileAppender. It might be a trade off between DRY and readability.

Reply:
I'll try it with the FileAppender and add some comments.